File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/net/packet/CBPacketBlockUpdate.java
package pize.tests.voxelgame.main.net.packet;

import pize.net.tcp.packet.IPacket;
import pize.net.tcp.packet.PacketHandler;
import pize.util.io.PizeInputStream;
import pize.util.io.PizeOutputStream;

import java.io.IOException;

public class CBPacketBlockUpdate extends IPacket<PacketHandler>{
    
    public static final int PACKET_ID = 10;
    
    public CBPacketBlockUpdate(){
        super(PACKET_ID);
    }
    
    public int x, y, z;
    public short state;
    
    public CBPacketBlockUpdate(int x, int y, int z, short state){
        this();
        
        this.x = x;
        this.y = y;
        this.z = z;
        this.state = state;
    }
    
    
    @Override
    protected void write(PizeOutputStream stream) throws IOException{
        stream.writeInt(x);
        stream.writeInt(y);
        stream.writeInt(z);
        stream.writeShort(state);
    }
    
    @Override
    public void read(PizeInputStream stream) throws IOException{
        x = stream.readInt();
        y = stream.readInt();
        z = stream.readInt();
        state = stream.readShort();
    }

}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/net/packet/CBPacketBlockUpdate.kt
package pize.tests.voxelgame.main.net.packet

import pize.net.tcp.packet.IPacket
import pize.net.tcp.packet.PacketHandler
import pize.util.io.PizeInputStream
import pize.util.io.PizeOutputStream
import java.io.IOException

class CBPacketBlockUpdate() : IPacket<PacketHandler?>(PACKET_ID) {
    var x = 0
    var y = 0
    var z = 0
    var state: Short = 0

    constructor(x: Int, y: Int, z: Int, state: Short) : this() {
        this.x = x
        this.y = y
        this.z = z
        this.state = state
    }

    @Throws(IOException::class)
    override fun write(stream: PizeOutputStream?) {
        stream!!.writeInt(x)
        stream.writeInt(y)
        stream.writeInt(z)
        stream.writeShort(state.toInt())
    }

    @Throws(IOException::class)
    override fun read(stream: PizeInputStream?) {
        x = stream!!.readInt()
        y = stream.readInt()
        z = stream.readInt()
        state = stream.readShort()
    }

    companion object {
        const val PACKET_ID = 10
    }
}
